lib/smart_app_launch/client_suite/registration_verification.rb

```diff
@@ -0,0 +1,61 @@
+module SMARTAppLaunch
+  module RegistrationVerification
+    def verify_registered_jwks(jwks_input)
+
+      jwks_warnings = []
+      parsed_smart_jwk_set = MockSMARTServer.jwk_set(smart_jwk_set, jwks_warnings)
+      jwks_warnings.each { |warning| add_message('warning', warning) }
+
+      # TODO: add key-specific verification per end of https://build.fhir.org/ig/HL7/smart-app-launch/client-confidential-asymmetric.html#registering-a-client-communicating-public-keys
+
+      unless parsed_smart_jwk_set.length.positive?
+        add_message(
+          'error',
+          'JWKS content for Confidential Asymmetric authentication does not include any valid keys.'
+        )
+      end
+    end
+
+    def verify_registered_launch_urls(launch_urls)
+      return unless launch_urls.present?
+
+      normalized_launch_urls = normalize_urls(launch_urls, 'launch URL')
+      output smart_launch_urls: normalized_launch_urls.join(',').strip
+    end
+
+    def verify_registered_redirect_uris(redirect_uris)
+      return unless redirect_uris.present?
+
+      normalized_redirect_uris = normalize_urls(redirect_uris, 'redirect URI')
+
+      output smart_redirect_uris: normalized_redirect_uris.join(',').strip
+    end
+
+    def normalize_urls(url_list, type_for_error)
+      normalized_urls = []
+      url_list.split(',').map { |one_url| one_url.strip }.each do |url|
```

Comment:
```suggestion
      url_list.split(',').map(&:strip).each_with_object([]) do |url, normalized_urls|
```
Use `each_with_object` if you're building up an object inside an `each` block. This will replace the initialization of `normalized_urls` above, as well as the return on line 58.

Reply:
fixed in [742af51](https://github.com/inferno-framework/smart-app-launch-test-kit/pull/96/commits/742af51aa15e06b30cc094a010658d3b287e036e)